aws-janitor/resources/internet_gateways.go

```diff
@@ -77,36 +74,46 @@ func (InternetGateways) MarkAndSweep(opts Options, set *Set) error {
 				break
 			}
 
-			var publicIPsToRelease []*string
+			if opts.DisassociatePublicIP {
+				var publicIPsToRelease []*string
 
-			pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
-				for _, eni := range page.NetworkInterfaces {
-					publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
+				pageFunc := func(page *ec2.DescribeNetworkInterfacesOutput, _ bool) bool {
+					for _, eni := range page.NetworkInterfaces {
+						publicIPsToRelease = append(publicIPsToRelease, eni.Association.PublicIp)
```

Comment:
I think that's the place where panic was happening originally: https://github.com/kubernetes-sigs/boskos/pull/158/files#r1083050091

Do we need to check that Association is not nil here?